post-scan-actions/fss-to-securityhub/lambda/index.js

```diff
@@ -0,0 +1,84 @@
+const AWS = require('aws-sdk');
+const uuid = require('uuid');
+
+parseDetectionPath = (path) => {
+  const bucket = path.substring(8, path.indexOf('.s3.'));
+  const key = path.split('/').slice(3).join('/');
+  return {
+    bucket,
+    key
+  };
+};
+
+publishToSecurityHub = async (securityhub, securityHubArn, accountId, malwareName, bucket, key ) => {
+  const date = new Date().toISOString();
+  const params = {
+    Findings : [
+      {
+        AwsAccountId: accountId,
+        CreatedAt: date,
+        Description: "Malware found!",
+        GeneratorId: "FSS",
+        Id: `${accountId}/${uuid.v4()}`,
+        ProductArn: securityHubArn,
+        SchemaVersion: "2018-10-08",
+        Resources: [
+          {
+            Id: key,
+            Type: "AwsS3Object"
+          },
+          {
+            Id: bucket,
+            Type: "AwsS3Bucket"
+          },
+        ],
+        Malware: [ 
+          { 
+              Name: malwareName,
+              Path: key,
+              State: "OBSERVED",
+              Type: "POTENTIALLY_UNWANTED"
+          }
+      ],
+        Severity: {
+          Label: "HIGH"
+        },
+        Title: "S3 Object had malware detected by FSS.",
+        Types: [
+          "Unusual Behaviors/Data"
+        ],
+        UpdatedAt: date
+      }
+    ]
+  };
+  try {
+    console.log(params);
+    const result = await securityhub.batchImportFindings(params).promise();
+    console.log(result);
+    return result;
+  } catch (error) {
+    console.error(error);
+    return error;
+  }
+};
+
+exports.handler = async (event) => {
+  const securityHubArn = process.env.SECURITY_HUB_ARN;
+  const accountId = process.env.ACCOUNT_ID;
+  const securityhub = new AWS.SecurityHub();
+  return Promise.all(event.Records.map(async (record) => {
+    console.log(record);
+    const message = JSON.parse(record.Sns.Message);
+    const findings = message.scanning_result.Findings;
+    
+    const results = [];
+    
+    if (findings) {
+      const malwareName = findings[0].malware;
```

Comment:
`findings` is an array. Please loop through all the finding unless you on purpose only want the first finding.

Reply:
If there are multiple malicious file in a zip.

Reply:
I really remember that conversation :-)